src/components/structures/RoomView.js

```diff
@@ -111,27 +116,25 @@ module.exports = React.createClass({
         // We can /peek though. If it fails then we present the join UI. If it
         // succeeds then great, show the preview (but we still may be able to /join!).
         if (!this.state.room) {
-            console.log("Attempting to peek into room %s", this.props.roomId);
-            MatrixClientPeg.get().peekInRoom(this.props.roomId).done(() => {
-                // we don't need to do anything - JS SDK will emit Room events
-                // which will update the UI. We *do* however need to know if we
-                // can join the room so we can fiddle with the UI appropriately.
-                var peekedRoom = MatrixClientPeg.get().getRoom(this.props.roomId);
-                if (!peekedRoom) {
-                    return;
-                }
-                var guestAccessEvent = peekedRoom.currentState.getStateEvents("m.room.guest_access", "");
-                if (!guestAccessEvent) {
-                    return;
-                }
-                if (guestAccessEvent.getContent().guest_access === "can_join") {
+            if (this.props.autoPeek) {
+                console.log("Attempting to peek into room %s", this.props.roomId);
+                MatrixClientPeg.get().peekInRoom(this.props.roomId).done(() => {
```

Comment:
You state in the description for `autoPeekDone` that it "track whether our autoPeek (if any) has completed)". I'm confused what this means in practice. Does this mean if it was *successful*? If it's just a flag for determining whether or not to show the loading spinner, then you need to change the `done()` to `finally()` to catch error conditions.